core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -577,28 +610,32 @@ class SnappySession(_sc: SparkContext) extends SparkSession(_sc) {
             // conn.close()
           }
         } while (!locked)
-        (conn, new TableIdentifier(table, Some(schemaName)))
+        Some(conn, new TableIdentifier(table, Some(schemaName)))
       case _ =>
         logDebug(s"Taking lock in " +
-            s" ${Thread.currentThread().getId} ")
+            s" ${Thread.currentThread().getId} and " +
+          s"app ${sqlContext.sparkContext.appName}")
         val regionLock = PartitionedRegion.getRegionLock(SnappySession.WRITE_LOCK_PREFIX + table,
           GemFireCacheImpl.getExisting)
-        regionLock.lock()
-        regionLock
+        regionLock.lock(serializedWriteLockTimeOut * 1000)
+        Some(regionLock)
     }
   }
 
   private[sql] def releaseLock(lock: Any): Unit = {
+    logInfo(s"Releasing the lock : $lock")
     lock match {
       case lock: RegionLock =>
         if (lock != null) {
-          logInfo(s"Going to unlock the lock object bulkOp $lock ")
+          logInfo(s"Going to unlock the lock object bulkOp $lock and " +
+            s"app ${sqlContext.sparkContext.appName}")
           lock.asInstanceOf[PartitionedRegion.RegionLock].unlock()
         }
       case (conn: Connection, id: TableIdentifier) =>
         var unlocked = false
         try {
-          logDebug(s"releasing lock on the server. ${id.table}")
+          logDebug(s"Going to unlock the lock on the server. ${id.table} for " +
+            s"app ${sqlContext.sparkContext.appName}")
```

Comment:
Can unlock fail? If yes then we should perhaps catch the exception and at least log a WARN. Right?